cmake/winml.cmake

```diff
@@ -415,10 +415,17 @@ set_target_properties(winml_dll
   PROPERTIES
   OUTPUT_NAME windows.ai.machinelearning)
 
-set_target_properties(winml_dll
-  PROPERTIES
-  LINK_FLAGS
-  "/DEF:${WINML_DIR}/windows.ai.machinelearning.def ${os_component_link_flags} /DELAYLOAD:d3d12.dll /DELAYLOAD:d3d11.dll /DELAYLOAD:dxgi.dll /DELAYLOAD:directml.dll")
+if (onnxruntime_USE_DML)
+  set_target_properties(winml_dll
+    PROPERTIES
+    LINK_FLAGS
+    "/DEF:${WINML_DIR}/windows.ai.machinelearning.def ${os_component_link_flags} /DELAYLOAD:d3d12.dll /DELAYLOAD:d3d11.dll /DELAYLOAD:dxgi.dll /DELAYLOAD:directml.dll")
+else()
+  set_target_properties(winml_dll
+    PROPERTIES
+    LINK_FLAGS
+    "/DEF:${WINML_DIR}/windows.ai.machinelearning.def ${os_component_link_flags} /DELAYLOAD:d3d12.dll /DELAYLOAD:d3d11.dll /DELAYLOAD:dxgi.dll")
+endif(onnxruntime_USE_DML)
```

Comment:
you can make this a list that you append to instead of trying to keep two lists in sync

Reply:
good point

Reply:
list(APPEND onnxruntime_EXTERNAL_DEPENDENCIES winml_adapter)


---
In reply to: [355070705](https://github.com/microsoft/onnxruntime/pull/2580#discussion_r355070705) [](ancestors = 355070705)